src/v/cloud_storage/tx_range_manifest.cc

```diff
@@ -12,20 +12,239 @@
 #include "bytes/iobuf.h"
 #include "bytes/iostream.h"
 #include "bytes/streambuf.h"
-#include "cloud_storage/partition_manifest.h"
 #include "cloud_storage/types.h"
 #include "container/fragmented_vector.h"
-#include "json/istreamwrapper.h"
 #include "model/record.h"
 
 #include <rapidjson/document.h>
+#include <rapidjson/error/en.h>
 #include <rapidjson/istreamwrapper.h>
 #include <rapidjson/ostreamwrapper.h>
 #include <rapidjson/rapidjson.h>
 #include <rapidjson/writer.h>
```

Comment:
Generally I would investigate the versions in `"json/"`, they have allocators that throw instead of returning null.